lib/SIL/IR/TypeLowering.cpp

```diff
@@ -3067,12 +3067,14 @@ bool TypeConverter::visitAggregateLeaves(
                            origTy.getPackExpansionPatternType(),
                            field, index);
       } else if (auto array = dyn_cast<BuiltinFixedArrayType>(ty)) {
-        auto origBFA = origTy.getAs<BuiltinFixedArrayType>();
-        insertIntoWorklist(
-            array->getElementType(),
-            AbstractionPattern(origTy.getGenericSignatureOrNull(),
-                               origBFA->getElementType()),
-            field, index);
+        auto origEltTy = AbstractionPattern::getOpaque();
+
+        if (auto origBFA = origTy.getAs<BuiltinFixedArrayType>()) {
```

Comment:
In other cases this is handled by adding a new method on AbstractionPattern that returns Opaque if `this` is Opaque, and otherwise does the destructuring, eg see `AbstractionPattern::getOptionalObjectType()`

Reply:
Actually, we don't reabstract the contents of an inline array do we? I think we should always be lowering the element type with the opaque pattern.

Try an InlineArray of function types or something.

Reply:
You're right, I think we should be using an opaque abstraction pattern here for elements always.